packages/nextra-theme-blog/src/ssg-context.tsx

```diff
@@ -0,0 +1,12 @@
+import { createContext, PropsWithChildren, useContext } from 'react'
+import React from 'react'
+const SsgContext = createContext<any>({})
+
+export const useSSG = () => useContext(SsgContext)
+
+export const SggProvider = ({
+  children,
+  ssg
+}: PropsWithChildren<{ ssg: any }>) => {
+  return <SsgContext.Provider value={ssg}>{children}</SsgContext.Provider>
```

Comment:
can `next/ssg` be used instead of this `ssg-context`?

Reply:
It can be used but it has to change the current layout implementation. I dont want to do that since it will also affact docs theme.

Currently `theme` cant access SSGContext value.
```
<BlogLayout>
  <SSGProvider/>
    <MDX/>
  </SSGProvider>
</BlogLayout>
```

Reply:
why this provider is necessary if we have already SSG provider here ?

https://github.com/shuding/nextra/blob/24a02f80452c061e325d002f4a35155154bba3ba/packages/nextra/src/loader.ts#L208-L212

Reply:
Because currently implementation is like this.
```mdx
<BlogLayout>
   <__nextra_SSGContext__.Provider value={props}> 
     <MDXContent /> 
   </__nextra_SSGContext__.Provider> 
</BlogLayout>
```